src/3d/chunks/qgschunkedentity_p.h

```diff
@@ -111,6 +113,20 @@ class QgsChunkedEntity : public Qt3DCore::QEntity
      */
     bool usingAditiveStrategy() const { return mAdditiveStrategy; }
 
+    /**
+     * Sets the limit of the GPU memory used to render the entity
+     * \since QGIS 3.24
+     */
+    void setGpuMemoryLimit( double gpuMemoryLimit ) { mGpuMemoryLimit = gpuMemoryLimit; }
+
+    /**
+     * Returns the limit of the GPU memory used to render the entity
+     * \since QGIS 3.24
+     */
+    double gpuMemoryLimit() const { return mGpuMemoryLimit; }
+
+    static double calculateEntityGpuMemorySize( Qt3DCore::QEntity *entity );
```

Comment:
It would be better to make this function virtual (and maybe pass its parent node?) - so that chunked entity implementations can reimplement it when necessary with some custom behavior. Also please add docstring